CMakeLists.txt

```diff
@@ -307,58 +307,57 @@ include(cmake/leveldb.cmake)
-if(NOT cxx_flags_overridden AND NOT CMAKE_CROSSCOMPILING)
-  add_library(warn_interface INTERFACE)
-  target_link_libraries(core_interface INTERFACE warn_interface)
-  if(MSVC)
-    try_append_cxx_flags("/W3" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("/wd4018" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("/wd4244" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("/wd4267" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("/wd4715" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("/wd4805" TARGET warn_interface SKIP_LINK)
-    target_compile_definitions(warn_interface INTERFACE
-      _CRT_SECURE_NO_WARNINGS
-      _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING
-    )
-  else()
-    try_append_cxx_flags("-Wall" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wextra" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wgnu" TARGET warn_interface SKIP_LINK)
-    # Some compilers will ignore -Wformat-security without -Wformat, so just combine the two here.
-    try_append_cxx_flags("-Wformat;-Wformat-security" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wvla" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wshadow-field" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wthread-safety" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wloop-analysis" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wredundant-decls" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wunused-member-function" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wdate-time" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wconditional-uninitialized" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wduplicated-branches" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wduplicated-cond" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wlogical-op" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Woverloaded-virtual" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wsuggest-override" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wimplicit-fallthrough" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wunreachable-code" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wdocumentation" TARGET warn_interface SKIP_LINK)
-
-    # Some compilers (gcc) ignore unknown -Wno-* options, but warn about all
-    # unknown options if any other warning is produced. Test the -Wfoo case, and
-    # set the -Wno-foo case if it works.
-    try_append_cxx_flags("-Wunused-parameter" TARGET warn_interface SKIP_LINK
-      IF_CHECK_PASSED "-Wno-unused-parameter"
-    )
-    try_append_cxx_flags("-Wself-assign" TARGET warn_interface SKIP_LINK
-      IF_CHECK_PASSED "-Wno-self-assign"
-    )
-  endif()
+if(NOT cxx_flags_overridden AND NOT CMAKE_TOOLCHAIN_FILE)
```

Comment:
Why is `AND NOT CMAKE_TOOLCHAIN_FILE` in here? The only case in master where we don't add these flags, is if cxxflags have been overridden, and that case should already be covered here?

Reply:
In the master branch, the `config.site` file uses the `CXXFLAGS` variable to pass flags from depends to the main build system, which makes it consider that flags are overridden.

In the CMake staging branch, the `CMAKE_CXX_FLAGS_INIT` variable is used for the same purpose.

Therefore, checking both `cxx_flags_overridden` and `CMAKE_TOOLCHAIN_FILE` exactly mirrors the master branch behavior, which will gone after https://github.com/bitcoin/bitcoin/pull/25972 anyway.

Reply:
I take it this is all going to be removed now?

Reply:
> I take it this is all going to be removed now?

Could you please to check the current PR branch @ eb51d1b65d03372e33094454674ef94151313b29 ?

It has been reworked during the recent rebase.